### PR TITLE
Added /pin/add endpoint call

### PR DESCRIPTION
**Pull request by icefoxen**
Hope I got this right!

### CODE REVIEW

File: ipfs-api/src/client.rs [hunk 1]
```diff
@@ -821,6 +821,19 @@ impl IpfsClient {
...
 
+    /// Pins a new object.
+    ///
+    /// Does not yet implement "progress" because reading it is kinda squirrelly.
+    ///
+    #[inline]
+    pub fn pin_add(
+        &self,
+        key: &str,
+        recursive: Option<bool>
```
Comment (ferristseng) [inline ipfs-api/src/client.rs]: Can you change this from `Option<bool>` type to just be a `bool` instead. I just made the change with some other calls!

**Comment by ferristseng** [comment]
Looks good!

I just recently changed API calls that took an `Option<bool>` to just use `bool` instead. I think it makes more sense to be explicit. The `None` scenario allowed the server to determine what to do, but I'd rather have the behaviour of each function call be consistent across IPFS versions.

I also added a small example of to each API call, and would like it if you could add a small one as well.

Thank you!!!

**Comment by icefoxen** [comment]
Will do!

**Comment by icefoxen** [comment]
Updates made.

**Comment by ferristseng** [comment]
Looks good, merging!

**Pull request merged**